apps/backend/src/filters/authentication-exception.filter.ts

```diff
@@ -1,11 +1,15 @@
 import {ArgumentsHost, Catch, ExceptionFilter} from '@nestjs/common';
+import {ConfigService} from '../config/config.service';
 
 @Catch(Error)
 export class AuthenticationExceptionFilter implements ExceptionFilter {
+  configService = new ConfigService();
   catch(exception: Error, host: ArgumentsHost): void {
     const ctx = host.switchToHttp();
     const response = ctx.getResponse();
-    response.cookie('authenticationError', exception.message);
+    response.cookie('authenticationError', exception.message, {
+      secure: this.configService.isInProductionMode()
+    });
```

Comment:
Here we are returning a Boolean based if we're running a production or other env.
I assume that when the @UseFilters(new AuthenticationExceptionFilter()) is called from the  authn.controller.ts and true is returned the line "this.authnService.login(req.user as User);" forces authentication?